api/internal/target/kusttarget_configplugin.go

```diff
@@ -223,6 +223,31 @@ var transformerConfigurators = map[builtinhelpers.BuiltinPluginType]func(
 	builtinhelpers.LabelTransformer: func(
 		kt *KustTarget, bpt builtinhelpers.BuiltinPluginType, f tFactory, tc *builtinconfig.TransformerConfig) (
 		result []resmap.Transformer, err error) {
+		for _, label := range kt.kustomization.Labels {
+			var c struct {
+				Labels     map[string]string
+				FieldSpecs []types.FieldSpec
+			}
+			c.Labels = label.Pairs
+			fss := types.FsSlice(label.FieldSpecs)
+			// merge the custom fieldSpecs with the default
+			if label.IncludeSelectors {
+				fss, err = fss.MergeAll(tc.CommonLabels)
```

Comment:
How about taking care of this in
https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/kustomization.go#L157
and never consulting `CommonLabels` from any other place?

If `CommonLabels` gets merged into `Labels` there, it also help the `fix` aspect of this change.

Reply:
I tried that but that will break a lot of commands and tests about commanLabels like `create` and `add`.

Reply:
sigh.  that's true.